frame/democracy/src/lib.rs

```diff
@@ -1232,13 +1193,26 @@ pub mod pallet {
 			let now = <frame_system::Pallet<T>>::block_number();
 			let (voting, enactment) = (T::VotingPeriod::get(), T::EnactmentPeriod::get());
 			let additional = if who == provider { Zero::zero() } else { enactment };
-			ensure!(now >= since + voting + additional, Error::<T>::TooEarly);
+			ensure!(
+				now >= since.saturating_add(voting).saturating_add(additional),
+				Error::<T>::TooEarly
+			);
```

Comment:
Not needed to add in this PR, but chrono has methods on time structs so that you can do:

```
ensure!( since.is_before(now), Error::<T>::TooEarly )
```

@PoisonPhang you reckon we should add an Extension trait to composable_support to implement this for `Moment`?